File: jAutoInvoice/src/java/net/sf/jautoinvoice/server/model/Fornecedor.java
/*
 * Fornecedor.java
 *
 * This file is part of jAutoInvoice, http://sourceforge.net/p/jautoinvoice
 *
 * Copyright (C) 2011 jAutoInvoice
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package net.sf.jautoinvoice.server.model;

import java.io.Serializable;

/**
 * 
 * @author Sérgio Lopes
 * @since 1.0
 */
public final class Fornecedor implements Serializable {

    private String id;
    private String nome;
    private String email;
    private String telefone;
    private String fax;
    private String notas;
    private boolean activo;

    public Fornecedor() {
        activo = true;
    }

    public Fornecedor(String id, String nome, String email, String telefone, String fax,
            String notas, boolean activo) {
        this.id = id;
        this.nome = nome;
        this.email = email;
        this.telefone = telefone;
        this.fax = fax;
        this.notas = notas;
        this.activo = activo;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFax() {
        return fax;
    }

    public void setFax(String fax) {
        this.fax = fax;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getNotas() {
        return notas;
    }

    public void setNotas(String notas) {
        this.notas = notas;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Fornecedor other = (Fornecedor) obj;

        if (this.id != null && this.id.equals(other.id)) {
            return true;
        }

        if ((this.nome == null) ? (other.nome != null) : !this.nome.equals(other.nome)) {
            return false;
        }
        if ((this.email == null) ? (other.email != null) : !this.email.equals(other.email)) {
            return false;
        }
        if ((this.telefone == null) ? (other.telefone != null) : !this.telefone.equals(other.telefone)) {
            return false;
        }
        if ((this.fax == null) ? (other.fax != null) : !this.fax.equals(other.fax)) {
            return false;
        }
        if (this.activo != other.activo) {
            return false;
        }
        return true;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 41 * hash + (this.id != null ? this.id.hashCode() : 0);
        hash = 41 * hash + (this.nome != null ? this.nome.hashCode() : 0);
        hash = 41 * hash + (this.email != null ? this.email.hashCode() : 0);
        hash = 41 * hash + (this.telefone != null ? this.telefone.hashCode() : 0);
        hash = 41 * hash + (this.fax != null ? this.fax.hashCode() : 0);
        hash = 41 * hash + (this.activo ? 1 : 0);
        return hash;
    }
}

File: jai/models/Fornecedor.php
<?php

/* This file is part of jAutoInvoice, a car workshop management software.
 * 
 * Copyright (c) 2012, Sérgio Lopes.
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 * 
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 * 
 * http://sourceforge.net/projects/jautoinvoice
 */

/**
 * @property int $idFornecedor
 * @property string $nome
 * @property string $email
 * @property string $notas
 * @property string $telemovel
 * @property string $telefone
 * @property string $website
 * @property string $morada
 * @property integer $activo
 *
 * The followings are the available model relations:
 * @property Email[] $emails
 * @property Material[] $materiais
 */
class Fornecedor extends CActiveRecord {

    /**
     * @return Fornecedor
     */
    public static function model($className = __CLASS__) {
        return parent::model($className);
    }

    public function tableName() {
        return '{{Fornecedor}}';
    }

    public function rules() {
        return array(
            array('nome', 'required'),
            array('nome, email, morada, website', 'length', 'max' => 255),
            array('telefone, telemovel', 'length', 'max' => 13),
            array('notas', 'safe'),
            // search
            array('nome, email', 'safe', 'on' => 'search'),
        );
    }

    public function relations() {
        return array(
            'emails' => array(self::MANY_MANY, 'Email', 'idFornecedor'),
            'materiais' => array(self::HAS_MANY, 'Material', 'MaterialFornecedor(idFornecedor,idMaterial)'),
        );
    }

    public function attributeLabels() {
        return array(
            'idFornecedor' => 'ID',
            'nome' => 'Nome',
            'email' => 'E-mail',
            'notas' => 'Notas',
            'telefone' => 'Telefone',
            'telemovel' => 'Telemóvel',
            'website' => 'Endereço WEB',
            'morada' => 'Morada'
        );
    }

    /**
     * @return CActiveDataProvider
     */
    public function search() {
        $criteria = new CDbCriteria();

        $criteria->compare('nome', $this->nome, true);
        $criteria->compare('email', $this->email, true);
        $criteria->compare('activo', 1);

        return new CActiveDataProvider($this, array(
                    'criteria' => $criteria,
                    'sort' => array(
                        'defaultOrder' => 'nome ASC',
                        )));
    }

}